odl/operator/operator.py

```diff
@@ -245,6 +518,74 @@ def __init__(self, domain, range, linear=False):
                 raise TypeError('range {!r} not a `LinearSpace` or `Field` '
                                 'instance.'.format(range))
 
+    def _call(self, x, out=None, **kwargs):
+        """Implementation of the operator evaluation.
+
+        This method is private backend for the evaluation of this
+        operator. It needs to match certain signature conventions,
+        and its implementation type is inferred from its signature.
+
+        The following signatures are allowed:
+
+        Python 2 and 3:
+            - ``_call(self, x)``  -->  out-of-place evaluation
+            - ``_call(self, vec, out)``  -->  in-place evaluation
+            - ``_call(self, x, out=None)``   --> both
+
+        Python 3 only:
+            - ``_call(self, x, *, out=None)`` (``out`` as keyword-only
+              argument)  --> both
+
+        For disambiguation, the instance name (the first argument) **must**
+        be 'self'.
+
+        The name of the ``out`` argument **must** be 'out', the second
+        argument may have any name.
+
+        Additional variable ``**kwargs`` and keyword-only arguments
+        (Python 3 only) are also allowed.
+
+        **General advice:**
+        - We recommend to implement the *in-place*
+          pattern if your code supports it since it avoids the
+          allocation of a new element and a copy compared to the
+          out-of-place one.
+        - If there is a significant performance gain from implementing
+          an out-of-place method separately, use the pattern for both
+          (``out`` optional) and decide according to the given ``out``
+          parameter which one to use.
+        - Use the out-of-place pattern only if your evaluation
+          code does not support in-place calculations.
+
+        See the documentation for more info on in-place vs. out-of-place
+        evaluation.
+
+        TODO: add link
+
+        Parameters
+        ----------
+        x : `Operator.domain`-like element
+            Element to which the operator is applied
+        out : `Operator.range` element, optional
+            Element to which the result is written
+        kwargs :
```

Comment:
No, and in fact, we shouldn't in general. I thought I had seen that style with NumPy, but they apparently just list their parameters as regular parameters in the list, not under some indented `kwargs` section. I'll quickly change that.

This case is somewhat special since we just propagate arguments.